src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -144,12 +151,12 @@ internal class TokenAcquisition : ITokenAcquisitionInternal
 
             try
             {
-                _application = await GetOrBuildConfidentialClientApplicationAsync().ConfigureAwait(false);
+                var application = await GetOrBuildConfidentialClientApplicationAsync().ConfigureAwait(false);
```

Comment:
>var application = await GetOrBuildConfidentialClientApplicationAsync().ConfigureAwait(false); [](start = 16, length = 93)

I think it makes sense to create a new CC object every time

Reply:
TokenAcquisition is a scoped service. We don't want to recreate it several times per request?

Reply:
depends on what we mean by scoped, my understanding is the current scope is "while handling everything triggered by the incomming call" and now you have multiple threads accessing the same object. MSAL can handle that for these scenarios, however I'm not sure what is best for guaranteeing the proper locks on accessing the cache. It may be either or, but if we include the resource in the obo cache I think the situation will be better if we have a CC object for each concurrent call. Probably worth testing out.

---
In reply to: [600316109](https://github.com/AzureAD/microsoft-identity-web/pull/1082#discussion_r600316109) [](ancestors = 600316109)